nibabel/cifti2/cifti2.py

```diff
@@ -1442,4 +1442,6 @@ def save(img, filename):
     filename : str
         filename to which to save image
     """
+    if img.nifti_header.get_intent()[0] == 'none':
+        raise ValueError("CIFTI image has an invalid intent code.")
```

Comment:
Even if we're not going to auto-set the intent code, I think this check should be in `Cifti2Image.to_file_map()`. That way it's hit whether you use `nb.cifti2.save`, `nb.save`, or `img.to_filename`.

Reply:
Also, you should exercise this code in the test. Use `assert_raises`. Easy to stick immediately before a `set_intent`, like:

```Python
with assert_raises(ValueError):
    img.to_filename(...)
img.nifti_header.set_intent(...)
img.to_filename(...)
```